homeassistant/components/qbittorrent/__init__.py

```diff
@@ -46,6 +46,23 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
 
     await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)
 
+    async def handle_get_torrents(service_call: ServiceCall):
+        coordinator: QBittorrentDataCoordinator = hass.data[DOMAIN][
+            config_entry.entry_id
+        ]
+        items = await coordinator.get_torrents(service_call.data[TORRENT_FILTER])
+        info = format_torrents(items)
+        return {
+            STATE_ATTR_TORRENT_INFO: info,
+        }
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_GET_TORRENTS,
+        handle_get_torrents,
+        supports_response=SupportsResponse.ONLY,
+    )
```

Comment:
I think our general consensus is now that we want this to be in `async_setup`, otherwise the service is dependent on the config entry.

I am also not sure how this exactly works when you have 2 config entries for qbittorrent

Reply:
Lemme take a look at this and check